assets/src/blocks/Timeline/useStyleSheet.js

```diff
@@ -0,0 +1,69 @@
+// useScript implementation from: https://usehooks.com/useScript/
+import { useEffect, useState } from 'react';
+
+// Hook
+let cachedStyleSheets = [];
+export const useStyleSheet = (href, media = 'all') => {
+
+  // Keeping track of script loaded and error state
+  const [state, setState] = useState({
```

Comment:
Another reason (which I'm not sure React mentions) is that with multiple `useState` calls, it is much easier to move a part of the state to a higher component, as you can move the entire call and don't need to disassemble the arguments and `useState` calls.

Reply:
Yes, I read that, but as the bold text says in that page: "we recommend to split state into multiple state variables based on which values tend to change together", and as these two (`loaded` and `error`) are always changing together, I think there's no need to split them here. It's just two, a very simple state, and they change together in all the useState calls.

And as the last paragraph in that question says: "Both putting all state in a single useState call, and having a useState call per each field can work. Components tend to be most readable when you find a balance between these two extremes, and group related state into a few independent state variables. If the state logic becomes complex, we recommend managing it with a reducer or a custom Hook.", and this is actually a custom Hook! I'm not planning to extract anything outside of it so far.

Reply:
Yeah there's definitely pros and cons to both approaches. I guess it doesn't matter as long as a single component doesn't use a combination of both.

Reply:
They do say "and group related state into a few independent state variables"...

In my previous comment I rather meant: if we use a hook with `[state, setState] = useState({a:b,c:d})`, and then need to add a second group of state variables that we want to keep separate, you would need to rename the first group to something more specific or it would get confusing.